pkg/config/config.go

```diff
@@ -334,6 +334,10 @@ func initConfig(config Config) {
 
 	// Metadata endpoints
 
+	// When enabled, hostname defined in the configuration (datadog.yaml) and starting with `ip-` or `domu` on EC2 is used as
+	// canonical hostname, otherwise the instance-id is used as canonical hostname.
+	config.BindEnvAndSetDefault("use_configuration_hostname_as_canonical_hostname", false)
```

Comment:
let's use a shorter config name, maybe `hostname_force_config_as_canonical`? WDYT?

Also, nit: can you move this close to the `hostname_fqdn` option in the code, just to group hostname-related stuff together?

Reply:
Renamed.
I put the config here as the usage is restricted to EC2 but it make more sense to move it near `hostname_fqdn`